### PR TITLE
[codex] Prevent overlapping schedule entries

### LINKED ISSUE #65: Überschneidung der Zeiten muss eine Fehlermeldung geben (siehe Foto)
Image

### PR DESCRIPTION
**Pull request by FleetAdmiralJakob**
## Summary

- Add server-side schedule conflict validation for manual day entries and learning-plan sessions.
- Return detailed German conflict errors that name the overlapping entry, date, and time range.
- Add Convex integration tests with Vitest/convex-test for overlap rejection, adjacent entries, learning-plan edits, and self-sync updates.

## Root Cause

Schedule writes were accepted independently across `dayEntries` and `learningPlans`, so overlapping time ranges for the same user/day could be created through multiple paths.

## Linked Issue

Closes #65

## Validation

- `pnpm test`
- `pnpm check`
- CodeRabbit CLI in WSL found two minor issues; both were addressed. Final rerun was blocked by the free CLI rate limit after the fixes.

## Summary by CodeRabbit

* **New Features**
  * Schedule conflict detection now blocks overlapping day entries and learning-plan sessions and surfaces descriptive error text with conflicting entry title and time range.
  * Entry creation UI shows inline error messages when scheduling conflicts occur.

* **Tests**
  * Added end-to-end tests covering conflict validation, edge-case day-key handling near midnight, and allowed boundary behaviors (adjacent times).

[Review Change Stack](https://app.coderabbit.ai/change-stack/Dayova/dayova-mvp/pull/88?utm_source=github_walkthrough&utm_medium=github&utm_campaign=change_stack)

### CODE REVIEW

**Comment by FleetAdmiralJakob** [comment]
image
image

File: convex/dayEntries.test.ts
Comment (Gamius00) [inline convex/dayEntries.test.ts:1]: This test file is a good Idea, but please describe the usage of the file in top of it.

Comment (Gamius00) [reply convex/dayEntries.test.ts:1]: Currently not important, but reminder for later

File: convex/scheduleConflicts.ts [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+	const variants = new Set([dayKey]);
+	const berlinDayKey = getBerlinDayKey(dayKey);
+	if (berlinDayKey) {
+		variants.add(berlinDayKey);
+		const match = dayKeyPattern.exec(berlinDayKey);
+		if (match) {
+			variants.add(
+				new Date(
+					Date.UTC(Number(match[1]), Number(match[2]) - 1, Number(match[3])),
+				).toISOString(),
```
[the remark this answers is not in the document]
Comment (Gamius00) [reply convex/scheduleConflicts.ts]: Sounds good, @FleetAdmiralJakob please check that, this is an potential Issue.
